### PR TITLE
Update DG - Overall Structure, Product Scope, NFR

### LINKED ISSUE #74: Update Developer Guide - Structure, Product Scope, NFR
Need to refine DG with better structure and also make edits to any areas in product scope and NFR which are outdated.

### PR DESCRIPTION
**Pull request by khoongwk**
Improvement to overall structure of DG to include table of contents and section numbering.

Minor edits to product scope and non-functional requirements to suit the new project direction.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -2,23 +2,50 @@
...
+the information related to their business, such as upcoming appointments, their list of clients, as well as profits recorded. 
+It focuses on the Command Line Interface (CLI) while providing users with a simple and clean Graphical User Interface (GUI).
+Thus, the main interaction with **GrAB3**. will be done through commands.
+
+The features of GrAB3 include:
+* Keeping track of existing/new clients 
+* Scheduling of upcoming appointments with clients
+* Recording of revenue from services provided
+* Recording of business-related expenses
+* Calculation of business profit.
```
Comment (yanlynnnnn) [inline docs/DeveloperGuide.md:37]: Think you might have added an extra '.' at the end!

**Review by yanlynnnnn — COMMENTED** [review]
LGTM 👍🏻

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (khoongwk) [reply docs/DeveloperGuide.md:37]: Resolved!

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -220,35 +247,38 @@ _{more aspects and alternatives to be added}_
...
+* Independent home-based business owner
+* Has little knowledge of business management software
+* Prefers to have relevant business components in a single application
 * Is a fast typist
 * Prefers typing over using mouse input
 
 **Value proposition**:
-* Consolidates everything a home-based buisness needs such as appointments, clients, expenses, revenue and services into one place
+* Consolidates everything a home-based business needs such as appointments, clients, expenses, revenue and services into one place.
+* Saves significant time for the business owner, who previously had to manage details across his/her contacts, excel sheets, and inventory notebooks.
```
Comment (RuiFengg) [inline docs/DeveloperGuide.md:277]: Some have full stops and some don't

**Review by RuiFengg — COMMENTED** [review]
Make all fullstops for DG to standardize

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (khoongwk) [reply docs/DeveloperGuide.md:277]: Resolved :)

**Review by RuiFengg — COMMENTED** [review]
LGTM

**Review by RuiFengg — APPROVED** [review]

**Pull request merged**